### ISSUE TITLE
Sometimes "Session is no longer valid" when leaving the bot connected to the VC for a few days

### ISSUE THREAD

**Issue by 2vg** [serenity-rs/songbird#76]
Songbird works fine in most cases, but if bot leave it connected to the VC channel for a few days, i will often get the following error:

```shell
ERROR songbird::driver::tasks: Catastrophic connection failure. Stopping.
Ws(WsClosed(Some(CloseFrame { code: Library(4006), reason: "Session is no longer valid." })))
 ```
 
 Do have any ideas for avoiding such errors?

**Comment by FelixMcFelix** [comment]
You've probably hit the maximum session length, so Discord have invalidated it -- this error message means that Discord closed your connection to them. Usually this comes up when you get moved channel or reconnect and the old WS thread is cleaned up.

The solution is to reconnect via a Call (i.e., get a new session from Discord's gateway).

-- Kyle

**Comment by 2vg** [comment]
@FelixMcFelix thanks!

**Comment by N1C0exe** [comment]
Hi @2vg have u found a workaround for this isssue ?

**Comment by 2vg** [comment]
@Nico-ITA 
for me, i solved it with this PR: https://github.com/serenity-rs/songbird/pull/81

**Comment by N1C0exe** [comment]
> @Nico-ITA for me, i solved it with this PR: #81

ah ok the problem is that im not using songbird :(
